core/templates/dev/head/domain/exploration/StateObjectFactory.js

```diff
@@ -17,14 +17,34 @@
  * domain objects.
  */
 
-oppia.factory('StateObjectFactory', [function() {
+oppia.factory('StateObjectFactory', [
+  'AnswerGroupObjectFactory',
+  function(AnswerGroupObjectFactory) {
   var State = function(name, content, interaction, paramChanges) {
     this.name = name;
     this.content = content;
-    this.interaction = interaction;
+    this.interaction = {
+      answer_groups:
+        generateAnswerGroupsFromBackend(interaction.answer_groups),
+      confirmed_unclassified_answers:
+        interaction.confirmed_unclassified_answers,
+      customization_args: interaction.customization_args,
+      default_outcome: interaction.default_outcome,
+      fallbacks: interaction.fallbacks,
+      id: interaction.id
+    };
     this.paramChanges = paramChanges;
   };
 
+  var generateAnswerGroupsFromBackend = function(answerGroupsBackendDict) {
+    var answerGroups = answerGroupsBackendDict.map(function(answerGroup) {
```

Comment:
answerGroup --> answerGroupDict (to distinguish it from the frontend domain object)